### PR TITLE
Pygeoapi: GRASS script add custom crs support and overall fixes

### PR DESCRIPTION
**Pull request by hblitza**
This addresses the GRASS script to calculate raster stats:
- fix grass modules import to make it working it again  
- add custom crs
- refactor input params (geojson, inputcrs)
- add check for cog url

@weskamm @JakobMiksch

### CODE REVIEW

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 1]
```diff
@@ -63,15 +106,25 @@
...
+        'polygonGeoJson': {
+            'title': 'Polygon GeoJSON',
+            'description': 'A polygon GeoJSON for which to compute zonal statistics of the COG',  # noqa: E501
             'minOccurs': 1,
-            'maxOccurs': 'unbounded',
+            'maxOccurs': 1,
             'schema': {
-                '$ref': 'http://schemas.opengis.net/ogcapi/features/part1/1.0/openapi/schemas/geometryGeoJSON.json'  # noqa: E501
+                '$ref': 'http://schemas.opengis.net/ogcapi/features/part1/1.0/openapi/schemas/geometryGeoJSON.json'
+                # noqa: E501
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:116]: the comment should be after the line to take effect I guess

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 2]
```diff
@@ -63,15 +106,25 @@
...
             }
-        }
+        },
+        'inputCrs': {
+            'title': 'Coordinate reference system',
+            'description': 'The coordinate reference system of the provided geometry. If not provided, crs of cog is assumed',  # noqa: E501
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:121]:
```suggestion
            'description': 'The coordinate reference system (CRS) of the provided geometry. If not provided, CRS of COG is assumed',  # noqa: E501
```

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 3]
```diff
@@ -93,11 +146,46 @@ def __init__(self, processor_def):  # noqa: D107
...
+        # create GeometryCollection
+        features = geoms["features"]
+        collection = GeometryCollection([shape(feature["geometry"]).buffer(0) for feature in features])
+
+        if input_crs is not None:
+            if isinstance(input_crs, str) and input_crs.startswith('EPSG:'):
+                cog_crs = get_crs_from_cog(cog_url)
+
+                if input_crs != cog_crs:
+
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:167]: remove line

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 4]
```diff
@@ -93,11 +146,46 @@ def __init__(self, processor_def):  # noqa: D107
...
+                raise Exception(
+                    'Provided CRS from user is not valid: {}'.format(input_crs)
+                )
 
-        result = generate_zonal_stats(rastermap=cog_url, geometries=geoms)
+        LOGGER.debug(f'Start to generate zonal stats for {cog_url}.')
+        try:
+            result = generate_zonal_stats(rastermap=cog_url, geometries=geoms, crs=cog_crs)
+        except Exception as e:
+            LOGGER.debug(f'Failed to generate zonal stats for {cog_url}.', e)
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:185]: does it work to have multiple arguments separated as comma?

File: pygeoapi/processes/algorithms/grass_algorithms.py [hunk 5]
```diff
@@ -28,85 +24,95 @@
         ["grass", "--config", "python_path"], text=True).strip()
 )
 
-# Import GRASS Python bindings
+# Import GRASS Python bindings after GRASS paths are set (important order)
```
Comment (JakobMiksch) [inline pygeoapi/processes/algorithms/grass_algorithms.py:27]:
```suggestion
# GRASS imports must be after GRASS paths are set (order is important)
```

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 6]
```diff
@@ -93,11 +146,46 @@ def __init__(self, processor_def):  # noqa: D107
...
+            if isinstance(input_crs, str) and input_crs.startswith('EPSG:'):
+                cog_crs = get_crs_from_cog(cog_url)
+
+                if input_crs != cog_crs:
+
+                    # reproject
+                    collection = reproject(collection, input_crs, cog_crs)
+                    # transform back to geojson
+                    geoms = mapping(collection)
+                else:
```
Comment (weskamm) [inline pygeoapi/processes/zonal_statistics_grass.py:172]: can be removed as it does not do anything

**Review by weskamm — APPROVED** [review]
Looking good to me!

**Comment by JakobMiksch** [comment]
@hblitza I fixed some basic linting [Fix lint](https://github.com/klips-project/klips-sdi/pull/144/commits/7740a49b5541ee67f5a10cf2c487995a010522d5)

File: pygeoapi/processes/zonal_statistics_grass.py (hunk 1 above)
Comment (JakobMiksch) [reply pygeoapi/processes/zonal_statistics_grass.py]: fixed in [Fix lint](https://github.com/klips-project/klips-sdi/pull/144/commits/7740a49b5541ee67f5a10cf2c487995a010522d5)

**Comment by hblitza** [comment]
Thanks for review!

**Pull request merged**